pkg/operator/operator.go

```diff
@@ -78,8 +82,61 @@ func (o *Operator) Run() error {
 		return fmt.Errorf("failed to start rgw. %+v", err)
 	}
 
+	err = o.createClientAccess(cluster)
+	if err != nil {
+		return fmt.Errorf("failed to create client access. %+v", err)
+	}
+
 	logger.Infof("DONE!")
 	<-time.After(1000000 * time.Second)
 
 	return nil
 }
+
+func (o *Operator) createClientAccess(clusterInfo *cephmon.ClusterInfo) error {
+	context := &clusterd.Context{}
+	conn, err := cephmon.ConnectToClusterAsAdmin(context, o.factory, clusterInfo)
+	if err != nil {
+		return fmt.Errorf("failed to connect to cluster: %+v", err)
+	}
+	defer conn.Shutdown()
+
+	// create the default rook pool
+	defaultPool := "rook"
+	pool := client.CephStoragePoolDetails{Name: defaultPool}
+	_, err = client.CreatePool(conn, pool)
+	if err != nil {
+		return fmt.Errorf("failed to create default rook pool: %+v", err)
+	}
+
+	// create a user for rbd clients
+	username := "client.rook-rbd-user"
+	access := []string{"osd", "allow rwx", "mon", "allow r"}
+
+	// get-or-create-key for the user account
+	rbdKey, err := client.AuthGetOrCreateKey(conn, username, access)
+	if err != nil {
+		return fmt.Errorf("failed to get or create auth key for %s. %+v", username, err)
+	}
+
+	// store the secret for the rbd user in the default namespace
+	secrets := map[string]string{
+		"key": rbdKey,
+	}
+	secret := &v1.Secret{
+		ObjectMeta: v1.ObjectMeta{Name: "rook-rbd-user"},
+		StringData: secrets,
+		Type:       k8sutil.RbdType,
+	}
+	_, err = o.clientset.Secrets(k8sutil.DefaultNamespace).Create(secret)
+	if err != nil {
+		if !k8sutil.IsKubernetesResourceAlreadyExistError(err) {
```

Comment:
@jbw976 +1. The rook operator should ensure that if its ran again (due to host shutdown, migration etc), that it will ensure the state of the rook cluster. And if ran again, nothing happens. So idempotency is desired. But that is probably not something we want to track on this bug issue. Perhaps just for the user-secrets